scripts/create_dataset.py

```diff
@@ -46,26 +46,323 @@ def init_dataset(dataset_name, dataset_description):
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    # validate the from table/column's exists in the input tables
+    validate_table_name_and_column_name(table_name=from_table_name,
+                                        column_name=from_column_name,
+                                        input_tables=input_tables)
+
+    # validate the to table/column's exists in the input tables
+    validate_table_name_and_column_name(table_name=to_table_name,
+                                        column_name=to_column_name,
+                                        input_tables=input_tables)
+
+
+def validate_table_name_and_column_name(table_name, column_name, input_tables):
+    """
+    Validates a relationship checking the the given table and column exist in input tables.
+    :param table_name: the name of input table containing has the input_column
+    :param column_name: the name of the input_column
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    input_columns = input_tables.get(table_name).get("columns")
+    if input_columns:
+        # columns is a list of dicts/columns containing multiple fields
+        for column in input_columns:
+            if column["name"] == column_name:
+                return
+
+    # raise in error if either the column or table name does not exist in the column names
+    raise SystemExit("Error validating table name and relationship: "
+                     "Column, {}, in table, {} does not exist"
+                     .format(column_name, table_name) + "\n")
+
+
+def init_relationship(from_table_name, from_column_name, to_table_name, to_column_name, array_of):
+    """
+    Creates a relationship given the input table/column name and the corresponding table/column name they relate to.
+    :param from_table_name: the name of input table containing the input_column
+    :param from_column_name: the name of the input_column
+    :param to_table_name: the name of table the input column relates to
+    :param to_column_name: the name of column the input column relates to
+    :param array_of: the value of the array_of of input column indicating if the column should be an array
+    """
+    relationship_name = "{}_{}_to_{}_{}".format(from_table_name, from_column_name, to_table_name, to_column_name)
+    cardinality = "one"
+
+    # if the column should be an arry of links the cardinality is one to many
+    if array_of is True:
+        cardinality = "many"
+    relationship = {
+        "name": relationship_name,
+        "from": {
+            "table": from_table_name,
+            "column": from_column_name,
+            "cardinality": "one"
+        },
+        "to": {
+            "table": to_table_name,
+            "column": to_column_name,
+            "cardinality": cardinality}
+    }
+
+    return relationship
```

Comment:
Instead of assigning the intermediate name, you can return the dictionary directly